### PR TITLE
Update to SudokuFuncs.py file

### PR DESCRIPTION
**Pull request by HASH-0021**
I made changes to some functions in 'SudokuFuncs.py' file. @Raj-Srikar, amend the changes accordingly.

### CODE REVIEW

File: SudokuFuncs.py [hunk 1]
```diff
@@ -188,63 +186,26 @@ def list_difference(li1, li2):
...
     """
-    Intitializes the 'combo_recursion' function, which will append the final list to the list, 'return_list'
+    This will append all combination lists to the list, 'return_list'
 
     args:
     -li - list, of which the combinations will be determined
 
     returns: list, 'return_list' which contains lists of all the possible combinations of 'li' list
     """
     return_list.clear()
```
Comment (Raj-Srikar) [inline SudokuFuncs.py:205]:
```suggestion
    return_list = []
```

Comment (Raj-Srikar) [inline SudokuFuncs.py]:
```suggestion
```

File: SudokuFuncs.py [hunk 2]
```diff
@@ -188,63 +186,26 @@ def list_difference(li1, li2):
...
-    -li1 - 1st list
-    -li2 - 2nd list
-
-    returns: resulting list after the intersection
-    """
-    inter = [value for value in li2 if value in li1]
-    return inter
-
-
 return_list = []        #Global declaration of the List to be returned after all the combinations are appended to it
```
Comment (Raj-Srikar) [inline SudokuFuncs.py]:
```suggestion
```

File: SudokuFuncs.py [hunk 3]
```diff
@@ -188,63 +186,26 @@ def list_difference(li1, li2):
...
     return_list.clear()
-    for i in li:
-        dup = li.copy()
-        ele = []
-        ele.append(dup.pop(dup.index(i)))
-        combo_recursion(dup,ele)
+    combination_in_tuple = list(itertools.permutations(li))
+    for i in combination_in_tuple:
+        return_list.append(list(i))
     return return_list
```
Comment (Raj-Srikar) [inline SudokuFuncs.py]:
```suggestion
    return list(map(lambda x: list(x), permutations(li)))
```
Edit the Docstring accordingly.

File: SudokuFuncs.py [hunk 4]
```diff
@@ -1,4 +1,5 @@
 import time
+import itertools
```
Comment (Raj-Srikar) [inline SudokuFuncs.py]:
```suggestion
from itertools import permutations
```

File: SudokuFuncs.py [hunk 5]
```diff
@@ -188,63 +186,26 @@ def list_difference(li1, li2):
 
     returns: resulting list after subtraction
     """
-    li_dif = [i for i in li1 + li2 if i not in li1 or i not in li2]
+    li_dif = [i for i in li1 + li2 if i in li1 and i not in li2]
     return li_dif
```
Comment (Raj-Srikar) [inline SudokuFuncs.py]:
```suggestion
    return [i for i in li1 + li2 if i in li1 and i not in li2]
```
Let's not waste memory by specifically allocating the list to a variable and then returning it ;)
Make changes for any other similar snippets in the program, if found.

File: SudokuFuncs.py [hunk 6]
```diff
@@ -52,7 +53,7 @@ def prompt_sudoku():
 def solveSudoku(fileName = "", showResults = False, showTime = False, matrix = []):
     """
     Solves a Sudoku by prompting about sudoku or reading a text file containing the sudoku or by directly
-    taking the matrix as an and either shows the solution or returns it. Can also tell the execution time
+    taking the matrix as variable and either shows the solution or returns it. Can also tell the execution time
     (Any one of the arguments 'fileName' and 'matrix' should be given. Else rises ValueError)
```
Comment (Raj-Srikar) [inline SudokuFuncs.py]:
```suggestion
    taking the matrix as a variable and either shows the solution or returns it. Can also tell the execution time
    (Any one of the arguments 'fileName' or 'matrix' should be given. Else rises ValueError)
```

File: SudokuFuncs.py [hunk 7]
```diff
@@ -52,7 +53,7 @@ def prompt_sudoku():
 def solveSudoku(fileName = "", showResults = False, showTime = False, matrix = []):
     """
     Solves a Sudoku by prompting about sudoku or reading a text file containing the sudoku or by directly
```
Comment (Raj-Srikar) [inline SudokuFuncs.py]:
```suggestion
    Solves a Sudoku by prompting the sudoku or reading a text file containing the sudoku or by directly
```

**Review by Raj-Srikar — CHANGES_REQUESTED** [review]
Looks Great 🔥! Address the suggestions I've made and update the changes accordingly. @HASH-0021

File: SudokuFuncs.py [hunk 8]
```diff
@@ -188,63 +186,26 @@ def list_difference(li1, li2):
...
-            dup2.append(dup1.pop(dup1.index(i)))
-            combo_recursion(dup1,dup2)
-    else:
-        return_list.append(li2+li1)
-
-
 def all_combinations(li):
     """
-    Intitializes the 'combo_recursion' function, which will append the final list to the list, 'return_list'
+    This will append all combination lists to the list, 'return_list'
```
Comment (HASH-0021) [inline SudokuFuncs.py]:
```suggestion
    This will return a list containing all combination lists
```

**Review by HASH-0021 — COMMENTED** [review]
Changed the docstrings for the 'all_combinations' function. No other changes found that are containing little-to-no-use variables or functions in 'SudokuFuncs.py' file. @Raj-Srikar, review this change and merge the pull request if everything is fine.

File: SudokuFuncs.py [hunk 9]
```diff
@@ -188,64 +186,21 @@ def list_difference(li1, li2):
...
 
 def all_combinations(li):
     """
-    Intitializes the 'combo_recursion' function, which will append the final list to the list, 'return_list'
+    This will append all combination lists to the list, 'return_list'
 
     args:
     -li - list, of which the combinations will be determined
 
     returns: list, 'return_list' which contains lists of all the possible combinations of 'li' list
```
Comment (Raj-Srikar) [inline SudokuFuncs.py]:
```suggestion
    returns: a list, which contains lists of all the possible combinations of 'li' list
```

**Review by Raj-Srikar — CHANGES_REQUESTED** [review]

**Comment by Raj-Srikar** [comment]
All good! Congrats on the first merged pull request. 🎉

**Pull request merged**